### PR TITLE
Correct encoding unsigned integers

### PR DESCRIPTION
**Pull request by it1804**
Values above 2^31-1 for unsigned integer types must be encoded in 5 bytes.

Signed-off-by: Ivan Torgashov <it1804@mail.ru>

### CODE REVIEW

**Comment by it1804** [comment]
For example, the ciscoResponseBytes in marshal_test.go is encoded by Gauge32 with the value 4294967295 in 5 bytes, not 4.
Unfortunately, I could not find this condition in the coding rules, but my test requests confirm this.
Traffic dumps confirm for:
0x7FFFFFFF:
```snmpset -v 1 -c private 192.168.100.124 .1.3.6.1.4.1.318.1.1.4.4.2.1.3.5 u 2147483647```
0x0080000000:
```snmpset -v 1 -c private 192.168.100.124 .1.3.6.1.4.1.318.1.1.4.4.2.1.3.5 u 2147483648```

Although snmpset generates the same five-byte values for INTEGER (i) in the case> 2147483647, I think this is incorrect. Documentary evidence is required.

**Review by TimRots — COMMENTED** [review]
Nice. Left some minor comments to improve error handling.

**Comment by TimRots** [comment]
Hi @it1804. Sorry for only getting back to this now. Would it be possible to resolve the merge conflict on this branch? Thinking this PR would be a good enhancement to add. Thanks in advance.

**Comment by it1804** [comment]
Hi! The conflict has been resolved.

File: helper_test.go [hunk 1]
```diff
@@ -71,6 +71,9 @@ var testsMarshalUint32 = []testsMarshalUint32T{
 	{65537, []byte{0x01, 0x00, 0x01}},          // FFFF + 2
 	{16777217, []byte{0x01, 0x00, 0x00, 0x01}}, // FFFFFF + 2
 	{18542501, []byte{0x01, 0x1a, 0xef, 0xa5}},
+	{18542501, []byte{0x01, 0x1a, 0xef, 0xa5}},
```
Comment (TimRots) [inline helper_test.go:74]: I think we should remove this duplicate entry for ``18542501``

File: helper.go [hunk 2]
```diff
@@ -358,24 +339,27 @@ func marshalUint32(v interface{}) ([]byte, error) {
...
-	// truncate leading zeros. Cleaner technique?
-	if source < 0x80 {
-		return bs[3:], nil
-	}
-	if source < 0x8000 {
-		return bs[2:], nil
+	buf := make([]byte, 4)
+	binary.BigEndian.PutUint32(buf, source)
+	i := 0
+	for ; i < 3; i++ {
```
Comment (TimRots) [inline helper.go:352]: It's probably more idiomatic to write this for loop as:
```go
for i := 0; i < 3; i++ {
```

**Review by TimRots — CHANGES_REQUESTED** [review]
Thank you @it1804. Left some minor nits and then this should be ready :)

**Comment by TimRots** [comment]
@SuperQ Can you give this PR a final review please? Thanks in advance.

**Review by SuperQ — APPROVED** [review]
LGTM

**Pull request merged**